daemon/graphdriver/overlay2/check.go

```diff
@@ -22,11 +23,23 @@ import (
 // directory or the kernel enable CONFIG_OVERLAY_FS_REDIRECT_DIR.
 // When these exist naive diff should be used.
 //
-// When running in a user namespace, returns errRunningInUserNS
-// immediately.
+// When running in a user namespace before kernel 5.11, returns
+// errRunningInUserNS immediately. In kernel 5.11 and later, we
+// check support as usual with some user namespace differences.
 func doesSupportNativeDiff(d string) error {
+	userxattr := false
 	if userns.RunningInUserNS() {
-		return errors.New("running in a user namespace")
+		if !kernel.CheckKernelVersion(5, 11, 0) {
+			return errors.New("running in a user namespace")
```

Comment:
I'm not sure if it's okay to just edit error messages or if that breaks stuff, but this error message should probably improved and suggest updating the kernel, like one of the error messages below.